features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -419,50 +421,57 @@ nsapi_error_t AT_CellularNetwork::disconnect()
 {
 #if NSAPI_PPP_AVAILABLE
     nsapi_error_t err = nsapi_ppp_disconnect(_at.get_file_handle());
+    if (err != NSAPI_ERROR_OK) {
```

Comment:
even if there is an error in ppp disconnect we should take control of filehandle to athandler and deactivate context.
Reasoning: ppp disconnect in any case releases filehandle